src/dmd/console.d

```diff
@@ -80,33 +80,59 @@ interface Console
 version (Windows)
 private final class WindowsConsole : Console
 {
+  nothrow:
+
   private:
     CONSOLE_SCREEN_BUFFER_INFO sbi;
     HANDLE handle;
     FILE* _fp;
 
+    static HANDLE getStdHandle(FILE *fp)
+    {
+        /* Determine if stream fp is a console
+         */
+        version (CRuntime_DigitalMars)
+        {
+            if (!isatty(fp._file))
+                return null;
+        }
+        else version (CRuntime_Microsoft)
+        {
+            if (!isatty(fileno(fp)))
+                return null;
+        }
+        else
+        {
+            static assert(0, "Unsupported Windows runtime.");
+        }
+
+        if (fp == stdout)
+            return GetStdHandle(STD_OUTPUT_HANDLE);
+        else if (fp == stderr)
+            return GetStdHandle(STD_ERROR_HANDLE);
+        else
+            return null;
+    }
+
   public:
 
     @property FILE* fp() { return _fp; }
 
-    this(FILE* fp)
+    static WindowsConsole create(FILE* fp)
```

Comment:
In my first commit I accidentally replaced `create` with a constructor. The last commit changes it back. A constructor is more idiomatic, I think.

Reply:
I just went with/kept the static factory function to have the is-compatible logic and according `return null` inside the class itself; no strong opinion about it though.

Reply:
The `null` thing is a bit odd. I think it would make more sense to have a no-op `Console` implementation which does nothing. That would allow removing all the `if (con)` checks throughout the codebase elsewhere. But that doesn't have to be done here.